plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/jdbc/JDBCExecutionContext.java

```diff
@@ -352,6 +351,15 @@ public boolean isAutoCommit()
         return autoCommit;
     }
 
+    private boolean isAutoCommit(boolean fallbackValue) {
+        try {
+            return isAutoCommit();
+        } catch (DBCException e) {
+            log.debug("Unable to determine if connection is in autocommit mode", e);
+            return fallbackValue;
+        }
+    }
```

Comment:
IMO it would be more eye-friendly if this method was inlined, so no obscure `fallbackValue` is passed back and forth.